### DISCUSSION TITLE
ofxProjectM

### DISCUSSION THREAD

**Discussion by Jonathhhan** [projectM-visualizer/projectm#853] [Show and tell]
I started to make a projectM addon for Open Frameworks (thanks for the very nice library). The new `projectm_opengl_render_frame_fbo` is really useful: https://github.com/Jonathhhan/ofxProjectM

**Comment by kblaschke** [comment]
Great to see another integration!

As discussed in #854, use the function `projectm_playlist_set_preset_switched_event_callback()` in `playlist_callbacks.h` [instead of this line](https://github.com/Jonathhhan/ofxProjectM/blob/c2926adcccf99a0a97df4d9a3c09702f47535296/src/ofxProjectM.cpp#L34) to get notified of successful preset changes. Also note here that if you call the `projectm_set_preset_switch_requested_event_callback` function at this point, you'll override the callback [previously registered by the playback library](https://github.com/projectM-visualizer/projectm/blob/9aa18cd9850833dc8131839821e494d7569b7620/src/playlist/PlaylistCWrapper.cpp#L32), effectively preventing it from changing presets.
